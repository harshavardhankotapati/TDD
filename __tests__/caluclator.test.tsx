import React from 'react';
import { fireEvent, render } from "@testing-library/react-native";
import Calculator from '../Caluclator/caluclator';
describe("Caluclator", () => {
    let mainComp = <Calculator />
    test('Calculator renders correctly', async () => {
        render(mainComp)
    });
    test('Calculator snapshot correctly', async () => {
        let snapShotCopy = render(mainComp).toJSON()
        expect(snapShotCopy).toMatchSnapshot()
    });
    test('Calculator query correctly', async () => {
        let getQuery = render(mainComp)
        const { getByText, getAllByText } = getQuery
        expect(getByText("1")).toBeTruthy()
        expect(getByText("2")).toBeTruthy()
        expect(getByText("3")).toBeTruthy()
        expect(getByText("4")).toBeTruthy()
        expect(getByText("5")).toBeTruthy()
        expect(getByText("6")).toBeTruthy()
        expect(getByText("7")).toBeTruthy()
        expect(getByText("8")).toBeTruthy()
        expect(getByText("9")).toBeTruthy()
        expect(getByText("+")).toBeTruthy()
        expect(getByText("=")).toBeTruthy()
        expect(getByText("/")).toBeTruthy()
        expect(getByText("*")).toBeTruthy()
        expect(getByText("-")).toBeTruthy()
        expect(getByText("C")).toBeTruthy()
        expect(getAllByText("0").length).toBeGreaterThan(0);
    });

  test('displays numbers when pressed', () => {
    const { getByText } = render(mainComp);

    fireEvent.press(getByText('5'));
    fireEvent.press(getByText('3'));

    expect(getByText('53')).toBeTruthy(); 
  });

  test('performs addition correctly', () => {
    const { getByText,getAllByText } = render(mainComp);
    fireEvent.press(getByText('5'));
    fireEvent.press(getByText('+'));
    fireEvent.press(getByText('3'));
    fireEvent.press(getByText('='));
    expect(getAllByText('8')).toBeTruthy(); 
  });

  test('performs subtraction correctly', () => {
    const { getByText ,getAllByText} = render(mainComp);

    fireEvent.press(getByText('9'));
    fireEvent.press(getByText('-'));
    fireEvent.press(getByText('4'));
    fireEvent.press(getByText('='));

    expect(getAllByText('5')).toBeTruthy(); 
  });

  test('performs multiplication correctly', () => {
    const { getByText } = render(mainComp);

    fireEvent.press(getByText('6'));
    fireEvent.press(getByText('*'));
    fireEvent.press(getByText('2'));
    fireEvent.press(getByText('='));

    expect(getByText('12')).toBeTruthy(); 
  });

  test('performs division correctly', () => {
    const { getByText,getAllByText } = render(mainComp);

    fireEvent.press(getByText('8'));
    fireEvent.press(getByText('/'));
    fireEvent.press(getByText('2'));
    fireEvent.press(getByText('='));

    expect(getAllByText('4')).toBeTruthy(); 
  });

  test('clears input when "C" is pressed', () => {
    const { getByText ,getAllByText} = render(mainComp);

    fireEvent.press(getByText('9'));
    fireEvent.press(getByText('C'));

    expect(getAllByText("0").length).toBeTruthy(); 
  });

  test('handles invalid input gracefully', () => {
    const { getByText } = render(mainComp);

    fireEvent.press(getByText('/'));
    fireEvent.press(getByText('='));

    expect(getByText('Error')).toBeTruthy(); 
  });
})

