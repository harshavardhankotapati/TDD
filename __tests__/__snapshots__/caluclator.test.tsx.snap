// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caluclator Calculator snapshot correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#222",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-end",
        "color": "white",
        "fontSize": 36,
        "marginBottom": 10,
        "marginRight": 20,
      }
    }
  >
    0
  </Text>
  <Text
    style={
      {
        "alignSelf": "flex-end",
        "color": "lightgreen",
        "fontSize": 28,
        "marginBottom": 20,
        "marginRight": 20,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        7
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        8
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        9
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        /
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        4
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        5
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        6
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        *
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        2
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        3
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        -
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        0
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        C
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        =
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#444",
          "borderRadius": 10,
          "margin": 5,
          "opacity": 1,
          "padding": 20,
          "width": 80,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
